### ISSUE TITLE
Softmax Loss for non-binary Labels

### ISSUE THREAD

**Issue by eggie5** [tensorflow/ranking#238]
I have multi-level relevance labels `y=[0,1,2,3]`, etc where `0` is irrelevant and `3` is most relevant. Typically, softmax x-ent you compare probably distributions and in practice the labels binary one-hot encoded and this is confirmed by the comment in the source: https://github.com/tensorflow/ranking/blob/1ee0788997510c6e0a76c57bca7d7864b71be130/tensorflow_ranking/python/losses.py#L427 

This would led me to think that I can't use my multi-level labels w/ the softmax ranking loss.

However, the referenced paper "An Analysis of the Softmax Cross Entropy Loss for Learning-to-Rank with Binary Relevance" by Bruch et al. and source implementation seem to normalize the labels: https://github.com/tensorflow/ranking/blob/1ee0788997510c6e0a76c57bca7d7864b71be130/tensorflow_ranking/python/losses_impl.py#L739

Just wanted to confirm that I can use unnormalized labels, like `y=[0,1,2,3]` w/ the softmax loss?

**Comment by xuanhuiwang** [comment]
Yes, you can use the weights y=[0,1,2,3] directly. Those are served as `weights` in the softmax loss so the results will be different from using y=[0, 1, 1, 1].
